SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -618,18 +618,16 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 
 	if (IsServer())
 	{
-		if (GlobalStateManager != nullptr)
+		if (GlobalStateManager != nullptr &&
+			!GlobalStateManager->IsReadyToCallBeginPlay() &&
```

Comment:
Isn't it a little odd that we check `!GlobalStateManager->IsReadyToCallBeginPlay()` and then straight after call `GlobalStateManager->SetCanBeginPlay(true);` and `GlobalStateManager->TriggerBeginPlay();`

Reply:
Well to be honest I wanted the check to be: !bCanBeginPlay
bCanBeginPlay is private on GSM with an accessor IsReadyToCallBeginPlay.

Reply:
Maybe I'm missing the deeper meaning here xD 
!CanBeginPlay implies you then call BeginPlay?

Reply:
If CanBeginPlay is false it means it seems valid to set it true with SetCanBeginPlay(true)